client/core/core.go

```diff
@@ -2078,6 +2092,7 @@ func (c *Core) prepareTrackedTrade(dc *dexConnection, form *TradeForm, crypter e
 	}
 	err = c.db.UpdateOrder(dbOrder)
 	if err != nil {
+		// Do NOT unlock the coins, they're already locked by server.
```

Comment:
Scenarios like this could cause the client's subsequent orders to get rejected by the server because the client (on restart) may attempt to use any of these coins to fund another order. Since the order details could not be persisted to db, the coins would not be re-locked on restart.

One thought is to detect server errors and attempt using a different set of coins to fund the new order. I mention this because aside from this scenario, @JoeGruffins and I had experienced situations where submitting orders fail with a coin locked error from the server.

Reply:
Since coins are no longer unlocked on client shutdown, the coins for this order will remain perpetually locked. The good news is the client will not attempt to re-use these coins to fund another order, which the DEX would have rejected. The bad news is that the coins will still be locked even after this order gets discarded by the DEX because the client did not respond to preimage requests. How to handle this? One thought is, since the order will not be tracked by the client, unlock the coins right away. If the client gets `coin is locked` errors from the DEX when trying to place future trades, those errors will be shortlived as the DEX will unlock those coins when the preimage request fails.

Reply:
Previous comment applies here as well:

The client does not save the order to db, nor will it respond to preimage requests for the order. Only advantage to keeping the coins locked is to prevent new orders from using these coins as they're already locked by the server. But we can't keep these locked forever. And there's no client-restart-agnostic reliable way of knowing that the server has also ditched the order, so we can safely unlock coins.

Any objections to unlocking right away?